Feature caliper base modifier

rfhaque opened the pull request:
Using base caliper (time mode) modifier

rfhaque committed:
Caliper modifier

rfhaque committed:
Caliper modifier

pearce8 committed:
Merge branch 'main' into feature_caliper_modifier

pearce8 committed:
Merge branch 'main' into feature_caliper_modifier

pearce8 committed:
Merge branch 'main' into feature_caliper_modifier

pearce8 committed:
Merge branch 'develop' into feature_caliper_modifier

rfhaque committed:
Merge remote-tracking branch 'origin/develop' into feature_caliper_modifier

rfhaque committed:
Merge with develop

pearce8 committed:
Merge branch 'develop' into feature_caliper_modifier

pearce8 committed:
Removing postprocessing

rfhaque committed:
modifier changes

pearce8 committed:
Merge branch 'develop' into feature_caliper_modifier

pearce8 committed:
Update benchpark

pearce8 committed:
Merge branch 'develop' into feature_caliper_modifier

Commit:
Support for topdown and cuda modes in caliper

Commit:
Merge with develop

pearce8 committed:
Merge branch 'develop' into feature_caliper_modifier

pearce8 committed:
Update license in modifier.py

Commit:
Merge remote-tracking branch 'origin/develop' into feature_caliper_modifier

pearce8 committed:
Moving modifiers to their own repository

Commit:
Fix modifier path and filename

Commit:
Fix caliper spack specs and remove elfutils

Commit:
Merge remote-tracking branch 'origin/develop' into feature_caliper_modifier

Commit:
Implementation of caliper top-down modifier

Commit:
Implementation of caliper cuda modifier

Commit:
Remove caliper-specific references from ramble.yaml

Commit:
Remove caliper-specific references from amg2023 cuda ramble.yaml

Commit:
Merge remote-tracking branch 'origin/develop' into feature_caliper_modifier

Commit:
Fix adiak calls

Commit:
Make separate modifier config file for each caliper modifier

Commit:
Merge with develop

Commit:
caliper for x86

Commit:
Fix benchpark script

Commit:
caliper cuda modifier for amg2023, saxpy

Commit:
Use caliper base modifier

Commit:
Build base mode without papi

Commit:
Remove external papi package

pearce8 commented on configs/LLNL-Magma-Penguin-icelake-OmniPath/caliper_topdown.yaml:
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+variables:
+  modifier_package_name: 'caliper-topdown'
```
Please remove from this PR, lets only have the platform independent modifier for now.

pearce8 commented on configs/LLNL-Sierra-IBM-power9-V100-Infiniband/caliper_cuda.yaml:
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+variables:
+  modifier_package_name: 'caliper-cuda'
```
Please remove from this PR, lets only have the platform independent modifier for now.

pearce8 commented on modifiers/caliper-cuda/modifier.py:
```diff
@@ -0,0 +1,18 @@
+from ramble.modkit import *
+from ramble.mod.benchpark.caliper import Caliper as CaliperBase
+
+class CaliperCuda(CaliperBase):
```
Please remove from this PR, lets only have the platform independent modifier for now.

pearce8 commented on modifiers/caliper-topdown/modifier.py:
```diff
@@ -0,0 +1,24 @@
+from ramble.modkit import *
+from ramble.mod.benchpark.caliper import Caliper as CaliperBase
+
+class CaliperTopdown(CaliperBase):
```
Please remove from this PR, lets only have the platform independent modifier for now.

pearce8 committed:
fixing lint style

Commit:
Remove topdown and cuda modifiers

Commit:
Merge branch 'feature_caliper_base_modifier' of github.com:LLNL/benchpark into feature_caliper_base_modifier

Commit:
Add modifier option to benchpark setup script, define caliper config once

Commit:
Remove cuda caliper modifier from amg2023 and saxpy

Commit:
Add base caliper modifier for amg2023(cuda,rocm) and saxpy(cuda,rocm)

pearce8 committed:
lint

pearce8 committed:
lint

pearce8 commented on modifiers/caliper/modifier.py:
```diff
@@ -0,0 +1,33 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from ramble.modkit import *
```
import * is triggering the CI failure.  Please see under actions/style/lint.  The solution is to import the specific modules you need instead of *.

slabasan replied:
Relaxed this requirement, consistent with ramble

pearce8 committed:
Merge branch 'develop' into feature_caliper_base_modifier

Commit:
Merge with develop

Commit:
Replace caliper.yaml with modifier.yaml

pearce8 committed:
Merge branch 'develop' into feature_caliper_base_modifier

Commit:
Add empty default modifier

Commit:
Merge branch 'feature_caliper_base_modifier' of github.com:LLNL/benchpark into feature_caliper_base_modifier

Commit:
Remove star import

slabasan committed:
relax flake8 for modifier functionality

ignores F403 and F405 in modifier.py

'from module import *' used; unable to detect undefined names (F403)
Name may be undefined, or defined from star imports: module (F405)

Commit:
Merge remote-tracking branch 'origin/develop' into feature_caliper_base_modifier

Commit:
Remove zero-modifier

Commit:
Merge with develop

pearce8 committed:
Merge branch 'develop' into feature_caliper_base_modifier

Commit:
Add FOM in AMG2023

pearce8 committed:
Merge branch 'develop' into feature_caliper_base_modifier
